### ISSUE TITLE
Cannot install bioconductor packages on local device

### ISSUE THREAD

**Issue by JaidipGill** [JaidipGill/supervised-single-cell#1]
# Attempt at installing GenomeInfoDbData (needed for Signac) from bioconductor
### 1. Responding 'all' to update package question
``` r
sessionInfo()
#> R version 4.2.1 (2022-06-23 ucrt)
#> Platform: x86_64-w64-mingw32/x64 (64-bit)
#> Running under: Windows 10 x64 (build 19045)
#> 
#> Matrix products: default
#> 
#> locale:
#> [1] LC_COLLATE=English_United Kingdom.utf8 
#> [2] LC_CTYPE=English_United Kingdom.utf8   
#> [3] LC_MONETARY=English_United Kingdom.utf8
#> [4] LC_NUMERIC=C                           
#> [5] LC_TIME=English_United Kingdom.utf8    
#> 
#> attached base packages:
#> [1] stats     graphics  grDevices utils     datasets  methods   base     
#> 
#> loaded via a namespace (and not attached):
#>  [1] digest_0.6.31     withr_2.5.0       R.methodsS3_1.8.2 lifecycle_1.0.3  
#>  [5] magrittr_2.0.3    reprex_2.0.2      evaluate_0.21     rlang_1.1.1      
#>  [9] cli_3.6.1         rstudioapi_0.14   fs_1.6.2          R.utils_2.12.2   
#> [13] R.oo_1.25.0       vctrs_0.6.3       styler_1.10.1     rmarkdown_2.22   
#> [17] tools_4.2.1       R.cache_0.16.0    glue_1.6.2        purrr_1.0.1      
#> [21] xfun_0.39         yaml_2.3.7        fastmap_1.1.1     compiler_4.2.1   
#> [25] htmltools_0.5.5   knitr_1.43
if (!require("BiocManager", quietly = TRUE))
  install.packages("BiocManager")
#> Warning: package 'BiocManager' was built under R version 4.2.3
#> Bioconductor version '3.16' is out-of-date; the current release version '3.17'
#>   is available with R version '4.3'; see https://bioconductor.org/install

BiocManager::install("GenomeInfoDbData")
#> 'getOption("repos")' replaces Bioconductor standard repositories, see 'help("repositories", package =
#> "BiocManager")' for details.
#> Replacement repositories:
#>    CRAN: https://cran.rstudio.com/
#> Bioconductor version 3.16 (BiocManager 1.30.21), R 4.2.1 (2022-06-23 ucrt)
#> Installing package(s) 'GenomeInfoDbData'
#> installing the source package ‘GenomeInfoDbData’

#> trying URL 'https://bioconductor.org/packages/3.16/data/annotation/src/contrib/GenomeInfoDbData_1.2.9.tar.gz'
#> Content type 'application/x-tar' length 11725198 bytes (11.2 MB)
#> downloaded 11.2 MB

#> The downloaded source packages are in
#>	‘C:\Users\jay_g\AppData\Local\Temp\RtmpoP9eHk\downloaded_packages’
#> Old packages: 'RcppArmadillo'
#> Update all/some/none? [a/s/n]: 
#> a

#>  There is a binary version available but the source version is later:
#>                  binary     source needs_compilation
#> RcppArmadillo 0.12.4.0.0 0.12.4.1.0              TRUE

#> installing the source package ‘RcppArmadillo’

#> trying URL 'https://cran.rstudio.com/src/contrib/RcppArmadillo_0.12.4.1.0.tar.gz'
#> Content type 'application/x-gzip' length 1373512 bytes (1.3 MB)
#> downloaded 1.3 MB


#> The downloaded source packages are in
#>	‘C:\Users\jay_g\AppData\Local\Temp\RtmpoP9eHk\downloaded_packages’
#> Warning messages:
#> 1: In install.packages(...) :
#>  installation of package ‘GenomeInfoDbData’ had non-zero exit status
#> 2: In install.packages(update[instlib == l, "Package"], l, repos = repos,  :
#>  installation of package ‘RcppArmadillo’ had non-zero exit status
```
### 2. Responding 'none' to update package question
```
#> Update all/some/none? [a/s/n]: 
#> n
#> Warning message:
#> In install.packages(...) :
#>   installation of package ‘GenomeInfoDbData’ had non-zero exit status
```

**Comment by brychan-az** [comment]
Have you tried installing Rcpp and RcppArmadillo outside of BiocManager? For example in fresh R session for the environment you plan to use run: install.packages("RcppArmadillo"). If that doesn't work you could try to download the binary files for windows form CRAN and try the install using those. It that case I would probably install (or update in already installed) Rcpp from a binary, then do the same for RcppArmadillo. 

Also just as a note, the install.packages and BiocManager functions have a "verbose" argument which you could set to true to get more information for debugging if you are still getting errors (e.g., BiocManager::install("GenomeInfoDbData", verbose = TRUE).

**Comment by JaidipGill** [comment]
I was able to install RcppArmadillo using install.packages. The issue seems to be just with BiocManager as GenomeInfoDbData still doesn't install. I wasn't able to see anything more useful from the verbose output below:
``` r
sessionInfo()
#> R version 4.2.1 (2022-06-23 ucrt)
#> Platform: x86_64-w64-mingw32/x64 (64-bit)
#> Running under: Windows 10 x64 (build 19045)
#> 
#> Matrix products: default
#> 
#> locale:
#> [1] LC_COLLATE=English_United Kingdom.utf8 
#> [2] LC_CTYPE=English_United Kingdom.utf8   
#> [3] LC_MONETARY=English_United Kingdom.utf8
#> [4] LC_NUMERIC=C                           
#> [5] LC_TIME=English_United Kingdom.utf8    
#> 
#> attached base packages:
#> [1] stats     graphics  grDevices utils     datasets  methods   base     
#> 
#> loaded via a namespace (and not attached):
#>  [1] digest_0.6.31     withr_2.5.0       R.methodsS3_1.8.2 lifecycle_1.0.3  
#>  [5] magrittr_2.0.3    reprex_2.0.2      evaluate_0.21     rlang_1.1.1      
#>  [9] cli_3.6.1         rstudioapi_0.14   fs_1.6.2          R.utils_2.12.2   
#> [13] R.oo_1.25.0       vctrs_0.6.3       styler_1.10.1     rmarkdown_2.22   
#> [17] tools_4.2.1       R.cache_0.16.0    glue_1.6.2        purrr_1.0.1      
#> [21] xfun_0.39         yaml_2.3.7        fastmap_1.1.1     compiler_4.2.1   
#> [25] htmltools_0.5.5   knitr_1.43
install.packages("RcppArmadillo")
#> Installing package into 'C:/Users/jay_g/AppData/Local/R/win-library/4.2'
#> (as 'lib' is unspecified)
#> package 'RcppArmadillo' successfully unpacked and MD5 sums checked
#> Warning: cannot remove prior installation of package 'RcppArmadillo'
#> Warning in file.copy(savedcopy, lib, recursive = TRUE): problem copying
#> C:\Users\jay_g\AppData\Local\R\win-library\4.2\00LOCK\RcppArmadillo\libs\x64\RcppArmadillo.dll
#> to
#> C:\Users\jay_g\AppData\Local\R\win-library\4.2\RcppArmadillo\libs\x64\RcppArmadillo.dll:
#> Permission denied
#> Warning: restored 'RcppArmadillo'
#> 
#> The downloaded binary packages are in
#>  C:\Users\jay_g\AppData\Local\Temp\RtmpMzkrnc\downloaded_packages
if (!require("BiocManager", quietly = TRUE))
  install.packages("BiocManager")
#> Warning: package 'BiocManager' was built under R version 4.2.3
#> Bioconductor version '3.16' is out-of-date; the current release version '3.17'
#>   is available with R version '4.3'; see https://bioconductor.org/install

BiocManager::install("GenomeInfoDbData", verbose = TRUE)
#> 'getOption("repos")' replaces Bioconductor standard repositories, see
#> 'help("repositories", package = "BiocManager")' for details.
#> Replacement repositories:
#>     CRAN: https://cran.rstudio.com/
#> Bioconductor version 3.16 (BiocManager 1.30.21), R 4.2.1 (2022-06-23 ucrt)
#> Installing package(s) 'GenomeInfoDbData'
#> installing the source package 'GenomeInfoDbData'
#> system (cmd0): C:/PROGRA~1/R/R-42~1.1/bin/x64/R CMD INSTALL
#> foundpkgs: GenomeInfoDbData, C:\Users\jay_g\AppData\Local\Temp\RtmpMzkrnc/downloaded_packages/GenomeInfoDbData_1.2.9.tar.gz
#> files: C:\Users\jay_g\AppData\Local\Temp\RtmpMzkrnc/downloaded_packages/GenomeInfoDbData_1.2.9.tar.gz
#> Warning in install.packages(...): installation of package 'GenomeInfoDbData'
#> had non-zero exit status
#> Old packages: 'geometries', 'labelled', 'Seurat', 'sfheaders', 'sp'
#> Update all/some/none? [a/s/n]: 
#> a

#>   There are binary versions available but the source versions are later:
#>            binary  source needs_compilation
#> geometries  0.2.2   0.2.3              TRUE
#> labelled   2.11.0  2.12.0             FALSE
#> Seurat      4.3.0 4.3.0.1              TRUE
#> sfheaders   0.4.2   0.4.3              TRUE
#> sp          1.6-1   2.0-0              TRUE

#> installing the source packages ‘geometries’, ‘labelled’, ‘Seurat’, ‘sfheaders’, ‘sp’

#> system (cmd0): C:/PROGRA~1/R/R-42~1.1/bin/x64/R CMD INSTALL
#> trying URL 'https://cran.rstudio.com/src/contrib/geometries_0.2.3.tar.gz'
#> Content type 'application/x-gzip' length 53624 bytes (52 KB)
#> downloaded 52 KB

#> trying URL 'https://cran.rstudio.com/src/contrib/labelled_2.12.0.tar.gz'
#> Content type 'application/x-gzip' length 238618 bytes (233 KB)
#> downloaded 233 KB

#> trying URL 'https://cran.rstudio.com/src/contrib/Seurat_4.3.0.1.tar.gz'
#> Content type 'application/x-gzip' length 1459200 bytes (1.4 MB)
#> downloaded 1.4 MB

#> trying URL 'https://cran.rstudio.com/src/contrib/sfheaders_0.4.3.tar.gz'
#> Content type 'application/x-gzip' length 84283 bytes (82 KB)
#> downloaded 82 KB

#> trying URL 'https://cran.rstudio.com/src/contrib/sp_2.0-0.tar.gz'
#> Content type 'application/x-gzip' length 1161401 bytes (1.1 MB)
#> downloaded 1.1 MB

#> foundpkgs: geometries, labelled, Seurat, sfheaders, sp,  
#> C:\Users\jay_g\AppData\Local\Temp\RtmpGGXwoD/downloaded_packages/geometries_0.2.3.tar.gz, 
#> C:\Users\jay_g\AppData\Local\Temp\RtmpGGXwoD/downloaded_packages/labelled_2.12.0.tar.gz, 
#> C:\Users\jay_g\AppData\Local\Temp\RtmpGGXwoD/downloaded_packages/Seurat_4.3.0.1.tar.gz, 
#> C:\Users\jay_g\AppData\Local\Temp\RtmpGGXwoD/downloaded_packages/sfheaders_0.4.3.tar.gz, #> 
#> C:\Users\jay_g\AppData\Local\Temp\RtmpGGXwoD/downloaded_packages/sp_2.0-0.tar.gz
#> files: C:\Users\jay_g\AppData\Local\Temp\RtmpGGXwoD/downloaded_packages/geometries_0.2.3.tar.gz, 
#> 	C:\Users\jay_g\AppData\Local\Temp\RtmpGGXwoD/downloaded_packages/labelled_2.12.0.tar.gz, 
#> 	C:\Users\jay_g\AppData\Local\Temp\RtmpGGXwoD/downloaded_packages/Seurat_4.3.0.1.tar.gz, 
#> 	C:\Users\jay_g\AppData\Local\Temp\RtmpGGXwoD/downloaded_packages/sfheaders_0.4.3.tar.gz, 
#> 	C:\Users\jay_g\AppData\Local\Temp\RtmpGGXwoD/downloaded_packages/sp_2.0-0.tar.gz

#> The downloaded source packages are in
#> 	‘C:\Users\jay_g\AppData\Local\Temp\RtmpGGXwoD\downloaded_packages’
#> Warning messages:
#> 1: In install.packages(...) :
#>   installation of package ‘GenomeInfoDbData’ had non-zero exit status
#> 2: In install.packages(update[instlib == l, "Package"], l, repos = repos,  :
#>   installation of package ‘geometries’ had non-zero exit status
#> 3: In install.packages(update[instlib == l, "Package"], l, repos = repos,  :
#>   installation of package ‘labelled’ had non-zero exit status
#> 4: In install.packages(update[instlib == l, "Package"], l, repos = repos,  :
#>   installation of package ‘Seurat’ had non-zero exit status
#> 5: In install.packages(update[instlib == l, "Package"], l, repos = repos,  :
#>   installation of package ‘sp’ had non-zero exit status
#> 6: In install.packages(update[instlib == l, "Package"], l, repos = repos,  :
#>   installation of package ‘sfheaders’ had non-zero exit status
```

<sup>Created on 2023-06-22 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by brychan-az** [comment]
It looks like each of those depend on Rcpp, so it might be work trying a fresh install of that using install.packages. Might also make sense to install each of those other packages that are failing using install.packages as well.

It also looks like BiocManager is using a cached download of the GenomeInfoDbData file at: C:\Users\jay_g\AppData\Local\Temp\RtmpMzkrnc/downloaded_packages/GenomeInfoDbData_1.2.9.tar.gz. Maybe try deleting that and installing with a fresh download.
